js/index.js

```diff
@@ -101,3 +103,165 @@ modalTriggers.forEach((trigger) => {
     });
   });
 });
+
+function popup(data) {
+  const isDeveloper = data.roles.developer;
+  const isDesigner = data.roles.designer;
+  const isMaven = data.roles.maven;
+  const isProductManager = data.roles.productmanager;
+  if (
+    data.incompleteUserDetails === false &&
+    !isDeveloper &&
+    !isDesigner &&
+    !isMaven &&
+    !isProductManager
+  ) {
+    const popupContainer = document.createElement('div');
+    popupContainer.className = 'roles-container';
+
+    const roleDetails = document.createElement('div');
+    roleDetails.className = 'role-details';
+
+    const roleHeading = document.createElement('h2');
+    roleHeading.className = 'role-heading';
+    roleHeading.textContent = 'Select your role';
+
+    const roleDetailsField = document.createElement('div');
+    roleDetailsField.className = 'role-details-field';
+
+    const roleButton = document.createElement('div');
+    roleButton.className = 'role-button';
+
+    const submitButton = document.createElement('button');
+    submitButton.textContent = 'Submit';
+
+    const spinner = document.createElement('span');
+    spinner.className = 'spinner';
+    submitButton.appendChild(spinner);
+
+    roleButton.appendChild(submitButton);
+
+    roleDetails.appendChild(roleHeading);
+    roleDetails.appendChild(roleDetailsField);
+    roleDetails.appendChild(roleButton);
+
+    popupContainer.appendChild(roleDetails);
+    popupContainer.style.display = 'none';
+    const mainElement = document.querySelector('main');
+    const firstChild = mainElement.firstChild;
+    mainElement.insertBefore(popupContainer, firstChild);
+
+    const roles = {};
+
+    const labels = [
+      { name: 'Developer', inputName: 'developer' },
+      { name: 'Designer', inputName: 'designer' },
+      { name: 'Maven', inputName: 'maven' },
+      { name: 'Product Manager', inputName: 'productmanager' },
+    ];
+
+    function createCheckbox(labelInfo) {
+      const label = document.createElement('label');
+      label.className = 'checkbox-label';
+
+      const input = document.createElement('input');
+      input.type = 'checkbox';
```

Comment:
is should be radio button. We are providing options to choose multiple roles @shubhamsigdar1?